### PR TITLE
Expose `DeviceCommissioner::ComputePASEVerifier` to Obj-C.

### PR DESCRIPTION
**Pull request by mburshteyn1**
#### Problem
* Expose `DeviceCommissioner::ComputePASEVerifier` to Obj-C.

#### Change overview
Added function`computePaseVerifier:iterations:salt:` to `MTRDeviceController` to expose `DeviceCommissioner ::ComputePASEVerifier` to Obj-C.

### CODE REVIEW

**Review by emargolis — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: src/darwin/Framework/CHIP/MTRDeviceController.mm [hunk 1]
```diff
@@ -668,6 +668,34 @@ - (void)setNocChainIssuer:(id<MTRNOCChainIssuer>)nocChainIssuer queue:(dispatch_
     });
 }
 
+- (nullable NSData *)computePaseVerifier:(uint32_t)setupPincode iterations:(uint32_t)iterations salt:(NSData *)salt
+{
+    __block CHIP_ERROR errorCode = CHIP_ERROR_INCORRECT_STATE;
+    if (![self isRunning]) {
+        [self checkForError:errorCode logMsg:kErrorNotRunning error:nil];
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/MTRDeviceController.mm:675]: Please use checkIsRunning.

File: src/darwin/Framework/CHIP/MTRDeviceController.mm [hunk 2]
```diff
@@ -668,6 +668,34 @@ - (void)setNocChainIssuer:(id<MTRNOCChainIssuer>)nocChainIssuer queue:(dispatch_
...
+        return nil;
+    }
+
+    __block NSData * result;
+    __block chip::Spake2pVerifier paseVerifier;
+    __block chip::ByteSpan saltByteSpan = chip::ByteSpan(static_cast<const uint8_t *>(salt.bytes), salt.length);
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/MTRDeviceController.mm:681]: Please use `AsByteSpan()`.

File: src/darwin/Framework/CHIP/MTRDeviceController.mm [hunk 3]
```diff
@@ -668,6 +668,34 @@ - (void)setNocChainIssuer:(id<MTRNOCChainIssuer>)nocChainIssuer queue:(dispatch_
...
+
+    dispatch_sync(_chipWorkQueue, ^{
+        if ([self isRunning]) {
+            errorCode = self.cppCommissioner->ComputePASEVerifier(iterations, setupPincode, saltByteSpan, paseVerifier);
+            MTR_LOG_ERROR("ComputePaseVerifier: %s", chip::ErrorStr(errorCode));
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/MTRDeviceController.mm:686]: And if this fails... now what?  We return garbage?

File: src/darwin/Framework/CHIP/MTRDeviceController.mm [hunk 4]
```diff
@@ -668,6 +668,34 @@ - (void)setNocChainIssuer:(id<MTRNOCChainIssuer>)nocChainIssuer queue:(dispatch_
...
+
+            uint8_t serializedVerifier[sizeof(paseVerifier.mW0) + sizeof(paseVerifier.mL)];
+            memcpy(serializedVerifier, paseVerifier.mW0, chip::kSpake2p_WS_Length);
+            memcpy(&serializedVerifier[sizeof(paseVerifier.mW0)], paseVerifier.mL, sizeof(paseVerifier.mL));
+
+            result = [NSData dataWithBytes:serializedVerifier length:sizeof(serializedVerifier)];
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/MTRDeviceController.mm:692]: Please use `AsData()`.

**Comment by bzbarsky-apple** [comment]
Also, why is this API on MTRDeviceController instances at all?  It does not use any instance state, either in practice or conceptually.  In the SDK API it's an instance API on DeviceCommissioner, but that's also an odd API choice and we should not be propagating it into the Darwin framework.

**Pull request merged**